## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 518451`
Commit: [`baa8589b62c79db768ffcf274363a673448732de`](https://github.com/NixOS/nixpkgs/commit/baa8589b62c79db768ffcf274363a673448732de) ([subsequent changes](https://github.com/NixOS/nixpkgs/compare/baa8589b62c79db768ffcf274363a673448732de..pull/518451/head))
Merge: [`b0918e47e72c4a31ed0393e32a644536cd3d45ad`](https://github.com/NixOS/nixpkgs/commit/b0918e47e72c4a31ed0393e32a644536cd3d45ad)

Logs: https://github.com/MiniHarinn/nixpkgs-review-gha/actions/runs/25607816372


---
### `x86_64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.sourmash</li>
    <li>python314Packages.sourmash.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>python313Packages.sourmash</li>
    <li>python313Packages.sourmash.dist</li>
    <li>tests.config-nix-unit</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-linux`</summary>
<details>
<summary>python314Packages.sourmash</summary>
<pre>        print(out)
        print(err)
    
        assert &quot;nifty: error: the following arguments are required: arg1&quot; in err
&gt;       assert &quot;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&quot; in err
E       AssertionError: assert &#x27;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&#x27; in &#x27;usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1\nsourmash scripts nifty: error: the following arguments are required: arg1\n&#x27;

tests/test_plugin_framework.py:385: AssertionError
----------------------------- Captured stdout call -----------------------------
running: sourmash in: /build/sourmashtest_1qzwnond
arguments [&#x27;sourmash&#x27;, &#x27;scripts&#x27;, &#x27;nifty&#x27;]

usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1
sourmash scripts nifty: error: the following arguments are required: arg1

=========================== short test summary info ============================
FAILED tests/test_plugin_framework.py::Test_EntryPointBasics_Command::test_cmd_3 - AssertionError: assert &#x27;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&#x27; in &#x27;usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1\nsourmash scripts nifty: error: the following arguments are required: arg1\n&#x27;
======= 1 failed, 3339 passed, 8 skipped, 1 xfailed in 194.44s (0:03:14) =======
mkdir -p failed for path /homeless-shelter/.config/matplotlib: [Errno 13] Permission denied: &#x27;/homeless-shelter&#x27;
Matplotlib created a temporary cache directory at /build/matplotlib-fss37cpq because there was an issue with the default path (/homeless-shelter/.config/matplotlib); it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.</pre>
</details>
</details>

---
### `aarch64-linux`
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.sourmash</li>
    <li>python314Packages.sourmash.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>python313Packages.sourmash</li>
    <li>python313Packages.sourmash.dist</li>
    <li>tests.config-nix-unit</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-linux`</summary>
<details>
<summary>python314Packages.sourmash</summary>
<pre>        print(out)
        print(err)
    
        assert &quot;nifty: error: the following arguments are required: arg1&quot; in err
&gt;       assert &quot;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&quot; in err
E       AssertionError: assert &#x27;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&#x27; in &#x27;usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1\nsourmash scripts nifty: error: the following arguments are required: arg1\n&#x27;

tests/test_plugin_framework.py:385: AssertionError
----------------------------- Captured stdout call -----------------------------
running: sourmash in: /build/sourmashtest__blk8e3c
arguments [&#x27;sourmash&#x27;, &#x27;scripts&#x27;, &#x27;nifty&#x27;]

usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1
sourmash scripts nifty: error: the following arguments are required: arg1

=========================== short test summary info ============================
FAILED tests/test_plugin_framework.py::Test_EntryPointBasics_Command::test_cmd_3 - AssertionError: assert &#x27;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&#x27; in &#x27;usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1\nsourmash scripts nifty: error: the following arguments are required: arg1\n&#x27;
======= 1 failed, 3339 passed, 8 skipped, 1 xfailed in 101.03s (0:01:41) =======
mkdir -p failed for path /homeless-shelter/.config/matplotlib: [Errno 13] Permission denied: &#x27;/homeless-shelter&#x27;
Matplotlib created a temporary cache directory at /build/matplotlib-8_nh0rdh because there was an issue with the default path (/homeless-shelter/.config/matplotlib); it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.</pre>
</details>
</details>

---
### `x86_64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 2 packages failed to build:</summary>
  <ul>
    <li>python314Packages.sourmash</li>
    <li>python314Packages.sourmash.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 3 packages built:</summary>
  <ul>
    <li>python313Packages.sourmash</li>
    <li>python313Packages.sourmash.dist</li>
    <li>tests.config-nix-unit</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python314Packages.sourmash</summary>
<pre>        print(out)
        print(err)
    
        assert &quot;nifty: error: the following arguments are required: arg1&quot; in err
&gt;       assert &quot;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&quot; in err
E       AssertionError: assert &#x27;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&#x27; in &#x27;usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1\nsourmash scripts nifty: error: the following arguments are required: arg1\n&#x27;

tests/test_plugin_framework.py:385: AssertionError
----------------------------- Captured stdout call -----------------------------
running: sourmash in: /nix/build/nix-14934-1809785680/sourmashtest_qu0rqhm5
arguments [&#x27;sourmash&#x27;, &#x27;scripts&#x27;, &#x27;nifty&#x27;]

usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1
sourmash scripts nifty: error: the following arguments are required: arg1

=========================== short test summary info ============================
FAILED tests/test_plugin_framework.py::Test_EntryPointBasics_Command::test_cmd_3 - AssertionError: assert &#x27;usage:  nifty [-h] [-q] [-d] [--other] [--do-fail] arg1&#x27; in &#x27;usage: sourmash scripts nifty [-h] [-q] [-d] [--other] [--do-fail] arg1\nsourmash scripts nifty: error: the following arguments are required: arg1\n&#x27;
======= 1 failed, 3339 passed, 8 skipped, 1 xfailed in 244.54s (0:04:04) =======
mkdir -p failed for path /homeless-shelter/.matplotlib: [Errno 30] Read-only file system: &#x27;/homeless-shelter&#x27;
Matplotlib created a temporary cache directory at /nix/build/nix-14934-1809785680/matplotlib-1lc3d54i because there was an issue with the default path (/homeless-shelter/.matplotlib); it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.</pre>
</details>
</details>

---
### `aarch64-darwin` (sandbox = relaxed)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python313Packages.sourmash</li>
    <li>python313Packages.sourmash.dist</li>
    <li>python314Packages.sourmash</li>
    <li>python314Packages.sourmash.dist</li>
  </ul>
</details>
<details>
  <summary>:white_check_mark: 1 package built:</summary>
  <ul>
    <li>tests.config-nix-unit</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python313Packages.sourmash</summary>
<pre>worker &#x27;gw14&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_index_best_containment[build_disk_revindex]&#x27;
================== xdist: maximum crashed workers reached: 12 ==================
=========================== short test summary info ============================
FAILED tests/test_cmd_index.py::test_index_signatures[rocksdb] - worker &#x27;gw0&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_index_signatures[rocksdb]&#x27;
FAILED tests/test_moltypes.py::test_index[skipm1n3-rocksdb] - worker &#x27;gw3&#x27; crashed while running &#x27;tests/test_moltypes.py::test_index[skipm1n3-rocksdb]&#x27;
FAILED tests/test_revindex.py::test_disk_revindex_prefetch_to_cg_colors_1 - worker &#x27;gw4&#x27; crashed while running &#x27;tests/test_revindex.py::test_disk_revindex_prefetch_to_cg_colors_1&#x27;
FAILED tests/test_index_protocol.py::test_index_location[build_disk_revindex] - worker &#x27;gw1&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_index_location[build_disk_revindex]&#x27;
FAILED tests/test_index_protocol.py::test_index_signatures_with_location[build_disk_revindex] - worker &#x27;gw2&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_index_signatures_with_location[build_disk_revindex]&#x27;
FAILED tests/test_cmd_index.py::test_gather_metagenome_abund[rocksdb] - worker &#x27;gw7&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_gather_metagenome_abund[rocksdb]&#x27;
FAILED tests/test_cmd_index.py::test_search_metagenome_downsample_containment[rocksdb] - worker &#x27;gw6&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_search_metagenome_downsample_containment[rocksdb]&#x27;
FAILED tests/test_index_protocol.py::test_index_manifest[build_disk_revindex] - worker &#x27;gw8&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_index_manifest[build_disk_revindex]&#x27;
FAILED tests/test_cmd_index.py::test_gather_metagenome_downsample[--prefetch---linear-rocksdb] - worker &#x27;gw9&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_gather_metagenome_downsample[--prefetch---linear-rocksdb]&#x27;
FAILED tests/test_moltypes.py::test_index[hp-rocksdb] - worker &#x27;gw10&#x27; crashed while running &#x27;tests/test_moltypes.py::test_index[hp-rocksdb]&#x27;
FAILED tests/test_cmd_index.py::test_gather_metagenome_threshold_bp_too_high[--linear---no-prefetch-rocksdb] - worker &#x27;gw11&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_gather_metagenome_threshold_bp_too_high[--linear---no-prefetch-rocksdb]&#x27;
FAILED tests/test_revindex.py::test_disk_revindex_load - worker &#x27;gw13&#x27; crashed while running &#x27;tests/test_revindex.py::test_disk_revindex_load&#x27;
FAILED tests/test_revindex.py::test_disk_revindex_manifest - worker &#x27;gw12&#x27; crashed while running &#x27;tests/test_revindex.py::test_disk_revindex_manifest&#x27;
FAILED tests/test_index_protocol.py::test_index_best_containment[build_disk_revindex] - worker &#x27;gw14&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_index_best_containment[build_disk_revindex]&#x27;
====== 14 failed, 3005 passed, 3 skipped, 1 xfailed in 163.20s (0:02:43) =======
mkdir -p failed for path /homeless-shelter/.matplotlib: [Errno 1] Operation not permitted: &#x27;/homeless-shelter&#x27;
Matplotlib created a temporary cache directory at /nix/build/nix-16881-2950364695/matplotlib-a49uuh64 because there was an issue with the default path (/homeless-shelter/.matplotlib); it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.</pre>
</details>
<details>
<summary>python314Packages.sourmash</summary>
<pre>worker &#x27;gw4&#x27; crashed while running &#x27;tests/test_revindex.py::test_disk_revindex_len&#x27;
================== xdist: maximum crashed workers reached: 12 ==================
=========================== short test summary info ============================
FAILED tests/test_cmd_index.py::test_index_signatures[rocksdb] - worker &#x27;gw0&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_index_signatures[rocksdb]&#x27;
FAILED tests/test_moltypes.py::test_index[skipm1n3-rocksdb] - worker &#x27;gw3&#x27; crashed while running &#x27;tests/test_moltypes.py::test_index[skipm1n3-rocksdb]&#x27;
FAILED tests/test_index_protocol.py::test_index_manifest[build_disk_revindex] - worker &#x27;gw1&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_index_manifest[build_disk_revindex]&#x27;
FAILED tests/test_cmd_index.py::test_search_metagenome_downsample_containment[rocksdb] - worker &#x27;gw2&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_search_metagenome_downsample_containment[rocksdb]&#x27;
FAILED tests/test_cmd_index.py::test_gather_save_matches_and_save_prefetch[--no-linear-rocksdb] - worker &#x27;gw5&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_gather_save_matches_and_save_prefetch[--no-linear-rocksdb]&#x27;
FAILED tests/test_revindex.py::test_disk_revindex_manifest - worker &#x27;gw7&#x27; crashed while running &#x27;tests/test_revindex.py::test_disk_revindex_manifest&#x27;
FAILED tests/test_moltypes.py::test_index[dayhoff-rocksdb] - worker &#x27;gw6&#x27; crashed while running &#x27;tests/test_moltypes.py::test_index[dayhoff-rocksdb]&#x27;
FAILED tests/test_index_protocol.py::test_index_best_containment_threshold_1[build_disk_revindex] - worker &#x27;gw8&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_index_best_containment_threshold_1[build_disk_revindex]&#x27;
FAILED tests/test_cmd_index.py::test_search_metagenome_downsample_index[rocksdb] - worker &#x27;gw10&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_search_metagenome_downsample_index[rocksdb]&#x27;
FAILED tests/test_cmd_index.py::test_gather_metagenome_abund[rocksdb] - worker &#x27;gw9&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_gather_metagenome_abund[rocksdb]&#x27;
FAILED tests/test_index_protocol.py::test_best_containment_threshold_5[build_disk_revindex] - worker &#x27;gw12&#x27; crashed while running &#x27;tests/test_index_protocol.py::test_best_containment_threshold_5[build_disk_revindex]&#x27;
FAILED tests/test_cmd_index.py::test_gather[--linear---prefetch-rocksdb] - worker &#x27;gw11&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_gather[--linear---prefetch-rocksdb]&#x27;
FAILED tests/test_cmd_index.py::test_gather_metagenome_threshold_bp[--linear---no-prefetch-rocksdb] - worker &#x27;gw13&#x27; crashed while running &#x27;tests/test_cmd_index.py::test_gather_metagenome_threshold_bp[--linear---no-prefetch-rocksdb]&#x27;
FAILED tests/test_revindex.py::test_disk_revindex_len - worker &#x27;gw4&#x27; crashed while running &#x27;tests/test_revindex.py::test_disk_revindex_len&#x27;
====== 14 failed, 2960 passed, 5 skipped, 1 xfailed in 157.53s (0:02:37) =======
mkdir -p failed for path /homeless-shelter/.matplotlib: [Errno 1] Operation not permitted: &#x27;/homeless-shelter&#x27;
Matplotlib created a temporary cache directory at /nix/build/nix-16881-2950364696/matplotlib-m5cfxzp3 because there was an issue with the default path (/homeless-shelter/.matplotlib); it is highly recommended to set the MPLCONFIGDIR environment variable to a writable directory, in particular to speed up the import of Matplotlib and to better support multiprocessing.</pre>
</details>
</details>

--- a/pkgs/development/python-modules/sourmash/default.nix
+++ b/pkgs/development/python-modules/sourmash/default.nix
@@ -4,6 +4,7 @@
   buildPythonPackage,
   stdenv,
   rustPlatform,
+  rocksdb_9_10,
   bitstring,
   cachetools,
   cffi,
@@ -18,6 +19,9 @@
   pyyaml,
   pytestCheckHook,
 }:
+let
+  rocksdb = rocksdb_9_10;
+in
 buildPythonPackage rec {
   pname = "sourmash";
   version = "4.9.4";
@@ -40,6 +44,11 @@ buildPythonPackage rec {
   ];
 
   buildInputs = [ iconv ];
+
+  env = {
+    ROCKSDB_INCLUDE_DIR = "${rocksdb}/include";
+    ROCKSDB_LIB_DIR = "${rocksdb}/lib";
+  };
 
   propagatedBuildInputs = [
     bitstring

--- a/pkgs/development/python-modules/sourmash/default.nix
+++ b/pkgs/development/python-modules/sourmash/default.nix
@@ -2,6 +2,7 @@
   lib,
   fetchPypi,
   buildPythonPackage,
+  pythonAtLeast,
   stdenv,
   rustPlatform,
   rocksdb_9_10,
@@ -74,6 +75,16 @@ buildPythonPackage rec {
     "test_compare_no_such_file"
     "test_do_sourmash_index_multiscaled_rescale_fail"
     "test_metagenome_kreport_out_fail"
+  ]
+  ++ lib.optionals (pythonAtLeast "3.14") [
+    # argparse subparser usage prefix changed in 3.14
+    "test_cmd_3"
+  ]
+  ++ lib.optionals (stdenv.hostPlatform.isDarwin && stdenv.hostPlatform.isAarch64) [
+    # rocksdb segfaults pytest workers
+    "rocksdb"
+    "disk_revindex"
+    "create_dataset_picklist"
   ];
 
   meta = {

